packages/catalog-process/test/db.test.ts

```diff
@@ -68,14 +117,1079 @@ describe("database test", async () => {
+        const descriptor: Descriptor = {
+          ...mockDescriptor,
+          state: descriptorState.suspended,
+        };
+        const eService: EService = {
+          ...mockEService,
+          descriptors: [descriptor],
+        };
+        await addOneEService(eService);
+        await catalogService.activateDescriptor(
+          eService.id,
+          descriptor.id,
+          getMockAuthData(eService.producerId)
+        );
+
+        const updatedDescriptor = {
+          ...descriptor,
+          state: descriptorState.published,
+        };
+
+        const writtenEvent = await readLastEventByStreamId(eService.id);
+        expect(writtenEvent.stream_id).toBe(eService.id);
+        expect(writtenEvent.version).toBe("1");
+        expect(writtenEvent.type).toBe("EServiceDescriptorUpdated");
+        const writtenPayload = decodeProtobufPayload({
+          messageType: EServiceDescriptorUpdatedV1,
+          payload: writtenEvent.data,
+        });
+        expect(writtenPayload.eServiceDescriptor).toEqual(
+          toDescriptorV1(updatedDescriptor)
+        );
+        expect(writtenPayload.eServiceId).toEqual(eService.id);
+      });
+
+      it("should throw eServiceNotFound if the eService doesn't exist", () => {
+        expect(
+          catalogService.activateDescriptor(
+            mockEService.id,
+            mockDescriptor.id,
+            getMockAuthData(mockEService.producerId)
+          )
+        ).rejects.toThrowError(eServiceNotFound(mockEService.id));
+      });
+
+      it("should throw operationForbidden if the requester is not allowed", async () => {
+        const descriptor: Descriptor = {
+          ...mockDescriptor,
+          state: descriptorState.suspended,
+        };
+        const eService: EService = {
+          ...mockEService,
+          descriptors: [descriptor],
+        };
+        await addOneEService(eService);
+        expect(
+          catalogService.activateDescriptor(
+            eService.id,
+            descriptor.id,
+            getMockAuthData()
+          )
+        ).rejects.toThrowError(operationForbidden);
+      });
+
+      it("should throw notValidDescriptor if the descriptor is in draft state", async () => {
+        const descriptor: Descriptor = {
+          ...mockDescriptor,
+          state: descriptorState.draft,
+        };
+        const eService: EService = {
+          ...mockEService,
+          descriptors: [descriptor],
+        };
+        await addOneEService(eService);
+        expect(
+          catalogService.activateDescriptor(
+            mockEService.id,
+            mockDescriptor.id,
+            getMockAuthData(eService.producerId)
+          )
+        ).rejects.toThrowError(
+          notValidDescriptor(descriptor.id, descriptorState.draft)
+        );
+      });
+    });
+
+    describe("clone descriptor", () => {
+      it("TO DO implement after understanding file manager", () => {
```

Comment:
what do you mean?

Reply:
I have to double-check how [this](https://github.com/pagopa/interop-be-monorepo/blob/main/packages/commons/src/fileManager.ts) works, because it is used in these endpoints. I want to make sure that the tests will not trigger api calls to the actual file manager service. Since this PR is already so huge, I thought of keeping this for a following PR.

Reply:
makes sense to me, let's just be sure that we do not forget it 👌

Reply:
Sure